sharpd/sharp_main.c

```diff
@@ -51,6 +51,7 @@
 DEFINE_MGROUP(SHARPD, "sharpd")
 
 zebra_capabilities_t _caps_p[] = {
+	ZCAP_NET_ADMIN, ZCAP_SYS_ADMIN, ZCAP_NET_RAW,
```

Comment:
Why does sharpd need these permissions?  This seems entirely different than calling fini right?

Reply:
having _no_ caps causes frr_fini to fail when it in turn finis the privs modules. I didn't want to completely re-work that path, since most daemons have some caps. these seemed like a fairly common set - but I don't think more than one is actually necessary. I could ... ask for something like ZCAP_BIND, or ZCAP_NICE - would that be ok?

Reply:
imo this is a question for @qlyoung or @eqvinox .  But I really suspect the answer is we should not be asking for permissions to do stuff we never intend to use as it becomes an attack vector.  I spent some time going through and pairing stuff down about 1.5 years ago( maybe a bit longer ) and I am having a hard time thinking how this could be ok.

Reply:
I've tweaked the privs/caps cleanup to tolerate no caps.